---
title: "Supplementary Information"
---

# Molecular Physiology

## Transcription

{{< video https://www.youtube.com/embed/bk7PW1FKMTI >}}

## Protein Degradation

{{< video https://www.youtube.com/embed/4DMqnfrzpKg >}}

## Epigenetics

### Basics

{{< video https://www.youtube.com/embed/_aAhcNjmvhc >}}

{{< video https://www.youtube.com/embed/Xjq5eEslJhw >}}

Webpage: [Why you should lick your rat](http://learn.genetics.utah.edu/content/epigenetics/rats/)


### Types

#### Histone Modification

Histone modifications are a key aspect of epigenetics, the study of how genes are regulated without altering the underlying DNA sequence. Histones are proteins around which DNA is wrapped in the cell nucleus. These proteins can be chemically modified in various ways, affecting how tightly or loosely the DNA is wound around them. These modifications include methylation, acetylation, phosphorylation, ubiquitination, and sumoylation, among others.

1. **Methylation:** The addition of methyl groups (CH3) to histones can either activate or repress gene expression, depending on where the methyl group is added. For instance, methylation of histone H3 at lysine 4 (H3K4me) is generally associated with gene activation, while methylation at H3K9 is linked with gene repression.

2. **Acetylation:** The addition of acetyl groups (COCH3) to histones typically loosens the DNA-histone interaction, making the DNA more accessible for transcription and thus promoting gene expression. Histone acetyltransferases (HATs) add acetyl groups, while histone deacetylases (HDACs) remove them.

3. **Phosphorylation:** The addition of phosphate groups (PO4) to histones can influence various cellular processes, including DNA repair, chromosome condensation during cell division, and gene expression.

4. **Ubiquitination and Sumoylation:** These modifications involve adding ubiquitin or SUMO (small ubiquitin-like modifier) proteins to histones, affecting DNA repair, transcriptional regulation, and chromatin organization.

Histone modifications regulate gene expression by altering the chromatin structure. This can either make the DNA more accessible to transcription factors and other proteins that promote transcription (thus activating gene expression) or make it less accessible (thereby repressing gene expression). These modifications do not change the DNA sequence but rather affect how cells interpret genes, contributing to the complexity of gene regulation in eukaryotic organisms. 

Overall, histone modifications are a dynamic and complex means of regulating gene expression, playing crucial roles in development, differentiation, and the response to environmental signals.

#### Short RNAs

Short RNAs, also known as small RNAs, are a class of non-coding RNA molecules that are typically 20-30 nucleotides long. They play crucial roles in gene regulation and maintenance of genome stability. The most well-known types of short RNAs include microRNAs (miRNAs), small interfering RNAs (siRNAs), and Piwi-interacting RNAs (piRNAs).

1. **MicroRNAs (miRNAs):** miRNAs are involved in post-transcriptional regulation of gene expression. They bind to complementary sequences on messenger RNA (mRNA) molecules, typically leading to the degradation of the mRNA or inhibition of its translation into proteins. miRNAs are key players in regulating various cellular processes including development, differentiation, cell proliferation, and apoptosis.

2. **Small Interfering RNAs (siRNAs):** siRNAs are typically involved in the RNA interference (RNAi) pathway. They originate from long double-stranded RNA precursors and are processed into siRNAs by an enzyme called Dicer. siRNAs are incorporated into the RNA-induced silencing complex (RISC), where they guide the complex to complementary mRNA targets, leading to mRNA cleavage and gene silencing. This mechanism is widely used in research to knock down the expression of specific genes.

3. **Piwi-interacting RNAs (piRNAs):** piRNAs are primarily found in animal cells and are associated with the Piwi subfamily of Argonaute proteins. They are involved in the silencing of transposable elements and other repetitive sequences in the genome, particularly in germ cells. This helps in protecting the integrity of the genome.

Apart from these, there are other classes of small RNAs, like small nucleolar RNAs (snoRNAs) and small nuclear RNAs (snRNAs), which are involved in processes like RNA modification (e.g., methylation) and splicing, respectively.

Overall, short RNAs are fundamental to the regulation of gene expression and maintenance of genomic integrity in cells. They play vital roles in a variety of biological processes and are also being studied for their potential in therapeutic applications, such as in the treatment of diseases like cancer and viral infections.


#### Long-non coding RNA

Long non-coding RNAs (lncRNAs) are a class of RNA molecules that are typically longer than 200 nucleotides and do not encode proteins. Despite not being translated into proteins, lncRNAs play significant roles in various cellular processes, and their functions are diverse and complex. Here are some key aspects of lncRNAs and their functions:

1. **Regulation of Gene Expression:** LncRNAs can regulate gene expression at multiple levels, including chromatin modification, transcription, and post-transcriptional processing. They can act as guides, scaffolds, decoys, or enhancers, depending on their interaction with DNA, RNA, or proteins.

    - **As Guides:** LncRNAs can direct ribonucleoprotein complexes to specific genomic loci to modify chromatin and regulate transcription.
    - **As Scaffolds:** They can bring together multiple proteins to form ribonucleoprotein complexes, influencing the activity and localization of these proteins.
    - **As Decoys:** LncRNAs can bind to and sequester proteins (like transcription factors or other RNA-binding proteins) away from their usual targets.
    - **As Enhancers:** Some lncRNAs can enhance gene expression by interacting with enhancer regions of the genome.

2. **Epigenetic Regulation:** LncRNAs are involved in epigenetic modifications like DNA methylation and histone modification, thereby influencing gene expression patterns without altering the underlying DNA sequence.

3. **Structural Organization:** They contribute to the structural organization of certain cellular compartments and can influence the assembly of macromolecular complexes.

4. **X Chromosome Inactivation:** In female mammals, the lncRNA Xist plays a crucial role in X chromosome inactivation, a process by which one of the X chromosomes is silenced to ensure dosage compensation.

5. **Alternative Splicing:** LncRNAs can modulate alternative splicing of pre-mRNAs, affecting the diversity of proteins produced in cells.

6. **Regulation of mRNA Stability and Translation:** Certain lncRNAs can influence the stability and translation of mRNAs, either by direct interaction or through the recruitment of other regulatory molecules.

7. **Cell Differentiation and Development:** LncRNAs are essential in numerous developmental processes, where they guide cell fate decisions and differentiation.

8. **Disease Association:** Dysregulation of lncRNAs has been linked to various diseases, including cancers, neurological disorders, and cardiovascular diseases.

Research into lncRNAs is still a rapidly evolving field. These RNAs are more numerous and diverse than initially thought, and they are increasingly recognized as key regulatory molecules in a wide range of biological processes and diseases. Their complexity and versatility make them important subjects for both basic biological research and potential therapeutic applications.

#### DNA Methylation

DNA methylation is a crucial epigenetic mechanism involved in regulating gene expression. It typically involves the addition of a methyl group (CH3) to the 5th carbon of the cytosine ring in DNA. This process predominantly occurs in regions of DNA where cytosine nucleotides are followed by guanine nucleotides, known as CpG sites. DNA methylation is catalyzed by enzymes called DNA methyltransferases (DNMTs).

The functional role of DNA methylation includes:

1. **Gene Regulation:** DNA methylation can either repress or activate gene expression. Methylation in gene promoter regions generally leads to gene silencing. This is because methylated DNA can inhibit the binding of transcription factors necessary for gene expression, or it can recruit proteins that condense the DNA, making it less accessible for transcription.

2. **Development and Differentiation:** DNA methylation patterns are crucial for normal development and cellular differentiation. As cells differentiate, specific genes are methylated or demethylated, which helps in the maintenance of cell-specific gene expression patterns.

3. **X-Chromosome Inactivation:** In female mammals, DNA methylation plays a critical role in X-chromosome inactivation, a process by which one of the two X chromosomes in females is silenced to ensure dosage compensation with males, who have only one X chromosome.

4. **Imprinting:** Genomic imprinting is a process where only one allele of a gene is expressed while the other is silenced. This silencing is often achieved through DNA methylation. Imprinted genes are crucial for growth and development, especially in the womb.

5. **Stability of the Genome:** DNA methylation helps in maintaining genome stability by suppressing the activity of transposable elements, which can move around the genome and cause mutations or chromosomal rearrangements if not properly controlled.

6. **Response to Environmental Factors:** Environmental factors can influence DNA methylation patterns, which in turn can affect gene expression. This is one way in which environmental factors can have a long-term impact on an organism's biology.

Alterations in DNA methylation patterns are associated with various diseases, including cancer, where abnormal hypermethylation of tumor suppressor genes or hypomethylation of oncogenes can occur. Research in this field continues to uncover the complex roles of DNA methylation in health and disease, making it a critical area of study in genetics and epigenetics.

### Ecology


**Epigenetics in Ecological and Evolutionary Processes**

Epigenetics plays a crucial role in the ecological and evolutionary processes of various organisms, particularly plants and animals. The interaction between epigenetics and the environment has been a subject of increasing interest (Feil & Fraga, 2012). Studies have shown that epigenetic mechanisms, such as DNA methylation and chromatin modifications, mediate responses to environmental cues and play a significant role in the adaptation of species to changing environmental conditions (Consuegra & López, 2016).

Furthermore, epigenetic diversity has been found to enhance the productivity and stability of plant populations, indicating its ecological implications (Latzel et al., 2013). Additionally, evidence from pyrosequencing suggests that natural variation in animal personality is associated with DNA methylation, highlighting the relevance of epigenetics in understanding behavioral consistency and its ecological and evolutionary implications (Verhulst et al., 2016).

The significance of epigenetics for plant ecology has been emphasized, with a call for increased transfer of knowledge and methods from model species research to genomes of evolutionarily divergent species, as well as the examination of responses to complex natural environments at a more mechanistic level (Richards et al., 2017). Furthermore, the opportunities and limitations of reduced representation bisulfite sequencing in plant ecological epigenomics have been explored, aiming to understand the unique contributions of epigenetic mechanisms to ecological and evolutionary processes (Paun et al., 2018).

Marine environmental epigenetics has also been studied, providing insights applicable to critical issues such as aquaculture, biomonitoring, and biological invasions, thereby improving the understanding and prediction of marine taxa responses to global climate change (Eirín‐López & Putnam, 2019).

In conclusion, epigenetics has emerged as a crucial factor in ecological and evolutionary processes, influencing the adaptation, productivity, and stability of plant populations, as well as the behavioral consistency of animals. The interaction between epigenetics and the environment has significant implications for understanding and predicting the responses of various organisms to environmental changes.

**References:**

- Consuegra, S. and López, C. (2016). *Epigenetic‐induced alterations in sex‐ratios in response to climate change: an epigenetic trap?*. Bioessays, 38(10), 950-958. [https://doi.org/10.1002/bies.201600058](https://doi.org/10.1002/bies.201600058)
- Eirín‐López, J. and Putnam, H. (2019). *Marine environmental epigenetics*. Annual Review of Marine Science, 11(1), 335-368. [https://doi.org/10.1146/annurev-marine-010318-095114](https://doi.org/10.1146/annurev-marine-010318-095114)
- Feil, R. and Fraga, M. (2012). *Epigenetics and the environment: emerging patterns and implications*. Nature Reviews Genetics, 13(2), 97-109. [https://doi.org/10.1038/nrg3142](https://doi.org/10.1038/nrg3142)
- Latzel, V., Allan, E., Silveira, A., Colot, V., Fischer, M., & Bossdorf, O. (2013). *Epigenetic diversity increases the productivity and stability of plant populations*. Nature Communications, 4(1). [https://doi.org/10.1038/ncomms3875](https://doi.org/10.1038/ncomms3875)
- Paun, O., Verhoeven, K., & Richards, C. (2018). *Opportunities and limitations of reduced representation bisulfite sequencing in plant ecological epigenomics*. New Phytologist, 221(2), 738-742. [https://doi.org/10.1111/nph.15388](https://doi.org/10.1111/nph.15388)
- Richards, C., Alonso, C., Becker, C., Bossdorf, O., Bucher, E., Colomé-Tatché, M., … & Verhoeven, K. (2017). *Ecological plant epigenetics: evidence from model and non‐model species, and the way forward*. Ecology Letters, 20(12), 1576-1590. [https://doi.org/10.1111/ele.12858](https://doi.org/10.1111/ele.12858)
- Verhulst, E., Mateman, A., Zwier, M., Samuel, P., Verhoeven, K., & Oers, K. (2016). *Evidence from pyrosequencing indicates that natural variation in animal personality is associated with drd4 dna methylation*. Molecular Ecology, 25(8), 1801-1811. [https://doi.org/10.1111/mec.13519](https://doi.org/10.1111/mec.13519)

---

### Invasive Species

Epigenetics plays a significant role in the expansion of invasive species, contributing to their adaptability and success in new environments. The rapid adaptation of invasive species, particularly in response to selection pressures generated by novel environments, involves epigenetic modulation as an adaptive strategy (Carneiro & Lyko, 2020). Epigenetic mechanisms, such as DNA methylation, have been suggested to compensate for losses in genetic variability and contribute to adaptive phenotypic plasticity in invasive species (Bock et al., 2015). Furthermore, epigenetic diversity has been proposed to facilitate the expansion of invasive species by acting as an alternative source of variation in areas with low genetic diversity (Sheldon et al., 2018). This plasticity acquired through epigenetic changes explains the ability of invasive species to expand and colonize new ecosystems, particularly in the presence of reduced genetic diversity due to founder effects (Ardura et al., 2017). Studies have also indicated that epigenetic mechanisms impact several phenotypic traits and may be important for the ecology and evolution of invasive species (Schrey et al., 2012). Additionally, epigenetic variation has been suggested to compensate for decreased genetic variation during introductions, highlighting its potential role in facilitating the success of invasive species in new environments (Schrey et al., 2012). The expansion of invasive species challenges our understanding of the process of adaptation, and epigenetic mechanisms are increasingly recognized as integral to the adaptation of species facing environmental changes (Chisholm et al., 2016). In conclusion, epigenetic mechanisms play a crucial role in the expansion of invasive species by facilitating rapid adaptation, compensating for genetic variability, and contributing to phenotypic plasticity. The plasticity acquired through epigenetic changes enables invasive species to thrive in new environments and colonize new ecosystems, ultimately contributing to their success as invasive species.

References:

- Ardura, A., Zaiko, A., Morán, P., Planes, S., & García‐Vázquez, E. (2017). Epigenetic signatures of invasive status in populations of marine invertebrates. Scientific Reports, 7(1). https://doi.org/10.1038/srep42193
- Bock, D., Caseys, C., Cousens, R., Hahn, M., Heredia, S., Hübner, S., ... & Rieseberg, L. (2015). What we still don't know about invasion genetics. Molecular Ecology, 24(9), 2277-2297. https://doi.org/10.1111/mec.13032
- Carneiro, V. and Lyko, F. (2020). Rapid epigenetic adaptation in animals and its role in invasiveness. Integrative and Comparative Biology, 60(2), 267-274. https://doi.org/10.1093/icb/icaa023
- Chisholm, R., Lorenzi, T., Desvillettes, L., & Hughes, B. (2016). Evolutionary dynamics of phenotype-structured populations: from individual-level mechanisms to population-level consequences. Zeitschrift Für Angewandte Mathematik Und Physik, 67(4). https://doi.org/10.1007/s00033-016-0690-7
- Consuegra, S. and López, C. (2016). Epigenetic‐induced alterations in sex‐ratios in response to climate change: an epigenetic trap?. Bioessays, 38(10), 950-958. https://doi.org/10.1002/bies.201600058
- Eirín‐López, J. and Putnam, H. (2019). Marine environmental epigenetics. Annual Review of Marine Science, 11(1), 335-368. https://doi.org/10.1146/annurev-marine-010318-095114
- Feil, R. and Fraga, M. (2012). Epigenetics and the environment: emerging patterns and implications. Nature Reviews Genetics, 13(2), 97-109. https://doi.org/10.1038/nrg3142
- Latzel, V., Allan, E., Silveira, A., Colot, V., Fischer, M., & Bossdorf, O. (2013). Epigenetic diversity increases the productivity and stability of plant populations. Nature Communications, 4(1). https://doi.org/10.1038/ncomms3875
- Paun, O., Verhoeven, K., & Richards, C. (2018). Opportunities and limitations of reduced representation bisulfite sequencing in plant ecological epigenomics. New Phytologist, 221(2), 738-742. https://doi.org/10.1111/nph.15388
- Richards, C., Alonso, C., Becker, C., Bossdorf, O., Bucher, E., Colomé-Tatché, M., ... & Verhoeven, K. (2017). Ecological plant epigenetics: evidence from model and non‐model species, and the way forward. Ecology Letters, 20(12), 1576-1590. https://doi.org/10.1111/ele.12858
- Schrey, A., Coon, C., Grispo, M., Awad, M., Imboma, T., McCoy, E., ... & Martin, L. (2012). Epigenetic variation may compensate for decreased genetic variation with introductions: a case study using house sparrows (passer domesticus) on two continents. Genetics Research International, 2012, 1-7. https://doi.org/10.1155/2012/979751
- Sheldon, E., Schrey, A., Andrew, S., Ragsdale, A., & Griffith, S. (2018). Epigenetic and genetic variation among three separate introductions of the house sparrow (passer domesticus) into Australia. Royal Society Open Science, 5(4), 172185. https://doi.org/10.1098/rsos.172185
- Verhulst, E., Mateman, A., Zwier, M., Samuel, P., Verhoeven, K., & Oers, K. (2016). Evidence from pyrosequencing indicates that natural variation in animal personality is associated with drd4 DNA methylation. Molecular Ecology, 25(8), 1801-1811. https://doi.org/10.1111/mec.13519


### Alternative Splicing


Lin, S., Yi, S. & Qiu, P. Comprehensive analysis of TCGA data reveals correlation between DNA methylation and alternative splicing. BMC Genomics 23, 758 (2022). https://doi.org/10.1186/s12864-022-08992-w






### Adaptation and Evolution

Yévenes, M., Gallardo-Escárate, C., & Gajardo, G. (2024). Epigenetic variation mediated by lncrnas accounts for adaptive genomic differentiation of the endemic blue mussel mytilus chilensis. Heliyon, 10(1), e23695. https://doi.org/10.1016/j.heliyon.2023.e23695




Lamka, G. F., Harder, A. M., Sundaram, M., Schwartz, T. S., Christie, M. R., DeWoody, J. A., … & Willoughby, J. R. (2022). Epigenetics in ecology, evolution, and conservation. Frontiers in Ecology and Evolution, 10. https://doi.org/10.3389/fevo.2022.871791

Abstract: Epigenetic variation is often characterized by modifications to DNA that do not alter the underlying nucleotide sequence, but can influence behavior, morphology, and physiological phenotypes by affecting gene expression and protein synthesis. In this review, we consider how the emerging field of ecological epigenetics (eco-epi) aims to use epigenetic variation to explain ecologically relevant phenotypic variation and predict evolutionary trajectories that are important in conservation. Here, we focus on how epigenetic data have contributed to our understanding of wild populations, including plants, animals, and fungi. First, we identified published eco-epi literature and found that there was limited taxonomic and ecosystem coverage and that, by necessity of available technology, these studies have most often focused on the summarized epigenome rather than locus- or nucleotide-level epigenome characteristics. We also found that while many studies focused on adaptation and heritability of the epigenome, the field has thematically expanded into topics such as disease ecology and epigenome-based ageing of individuals. In the second part of our synthesis, we discuss key insights that have emerged from the epigenetic field broadly and use these to preview the path toward integration of epigenetics into ecology. Specifically, we suggest moving focus to nucleotide-level differences in the epigenome rather than whole-epigenome data and that we incorporate several facets of epigenome characterization (e.g., methylation, chromatin structure). Finally, we also suggest that incorporation of behavior and stress data will be critical to the process of fully integrating eco-epi data into ecology, conservation, and evolutionary biology. 


Venney, C. J., Anastasiadi, D., Wellenreuther, M., & Bernatchez, L. (2023). The evolutionary complexities of dna methylation in animals: from plasticity to genetic evolution. Genome Biology and Evolution, 15(12). https://doi.org/10.1093/gbe/evad216

The importance of DNA methylation in plastic responses to environmental change and evolutionary dynamics is increasingly recognized. Here, we provide a Perspective piece on the diverse roles of DNA methylation on broad evolutionary timescales, including (i) short-term transient acclimation, (ii) stable phenotypic evolution, and (iii) genomic evolution. We show that epigenetic responses vary along a continuum, ranging from short-term acclimatory responses in variable environments within a generation to long-term modifications in populations and species. DNA methylation thus unlocks additional potential for organisms to rapidly acclimate to their environment over short timeframes. If these changes affect fitness, they can circumvent the need for adaptive changes at the genome level. However, methylation has a complex reciprocal relationship with genetic variation as it can be genetically controlled, yet it can also induce point mutations and contribute to genomic evolution. When habitats remain constant over many generations, or populations are separated across habitats, initially plastic phenotypes can become hardwired through epigenetically facilitated mutagenesis. It remains unclear under what circumstances plasticity contributes to evolutionary outcomes, and when plastic changes will become permanently encoded into genotype. We highlight how studies investigating the evolution of epigenetic plasticity need to carefully consider how plasticity in methylation state could evolve among different evolutionary scenarios, the possible phenotypic outcomes, its effects on genomic evolution, and the proximate energetic and ultimate fitness costs of methylation. We argue that accumulating evidence suggests that DNA methylation can contribute toward evolution on various timescales, spanning a continuum from acclimatory plasticity to genomic evolution.
