<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Epigenetic Phenomena Connecting Climate Change and Coastal Marine Species - DNA Methylation Assessment</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Epigenetic Phenomena Connecting Climate Change and Coastal Marine Species</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools ms-auto">
    <a href="https://github.com/sr320/course-ASI-2024" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../modules/DRAFT_DNA_methylation.html">Tutorials</a></li><li class="breadcrumb-item"><a href="../modules/DRAFT_DNA_methylation.html">DNA Methylation Assessment</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Logistics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../modules/day01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Agenda</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">Computation Basics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../modules/00-markdown.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">markdown</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../modules/01-github.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">GitHub</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../modules/02-bash.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">bash</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../modules/03-rstudio.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">RStudio</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Biology and Ecology</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../modules/slidedecks.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Slidedecks</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../modules/10-climate.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Climate Change</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../modules/20-supp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Molecular Physiology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../modules/21-supp.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Epigenetics : Basics</span></a>
  </div>
</li>
          <li class="sidebar-item">
 <span class="menu-text">modules/22-supp.qmd</span>
  </li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true">
 <span class="menu-text">Tutorials</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../modules/DRAFT_DNA_methylation.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">DNA Methylation Assessment</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true">
 <span class="menu-text">Computer Labs</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assignments/00-bash.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">bash</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assignments/01-blast.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">NCBI Blast</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assignments/02-DGE.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Differential Gene Expression</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assignments/01-blast.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">NCBI Blast</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../assignments/02-DGE.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Differential Gene Expression</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
  <li><a href="#background" id="toc-background" class="nav-link" data-scroll-target="#background">Background</a></li>
  <li><a href="#experimental-considerations" id="toc-experimental-considerations" class="nav-link" data-scroll-target="#experimental-considerations">Experimental Considerations</a>
  <ul class="collapse">
  <li><a href="#sample-size" id="toc-sample-size" class="nav-link" data-scroll-target="#sample-size">Sample Size</a></li>
  <li><a href="#sequencing-depth" id="toc-sequencing-depth" class="nav-link" data-scroll-target="#sequencing-depth">Sequencing Depth</a></li>
  <li><a href="#bisulfite-conversion-efficiency" id="toc-bisulfite-conversion-efficiency" class="nav-link" data-scroll-target="#bisulfite-conversion-efficiency">Bisulfite Conversion Efficiency</a></li>
  </ul></li>
  <li><a href="#tutorial-details" id="toc-tutorial-details" class="nav-link" data-scroll-target="#tutorial-details">Tutorial Details</a></li>
  </ul></li>
  <li><a href="#sequence-quality" id="toc-sequence-quality" class="nav-link" data-scroll-target="#sequence-quality">Sequence Quality</a></li>
  <li><a href="#read-alignment" id="toc-read-alignment" class="nav-link" data-scroll-target="#read-alignment">Read Alignment</a>
  <ul class="collapse">
  <li><a href="#software-considerations" id="toc-software-considerations" class="nav-link" data-scroll-target="#software-considerations">Software Considerations</a></li>
  <li><a href="#genome-preparation" id="toc-genome-preparation" class="nav-link" data-scroll-target="#genome-preparation">Genome Preparation</a></li>
  <li><a href="#aligning-reads" id="toc-aligning-reads" class="nav-link" data-scroll-target="#aligning-reads">Aligning reads</a>
  <ul class="collapse">
  <li><a href="#optimizing-alignment-sensitivity" id="toc-optimizing-alignment-sensitivity" class="nav-link" data-scroll-target="#optimizing-alignment-sensitivity">Optimizing alignment sensitivity</a></li>
  </ul></li>
  <li><a href="#deduplication" id="toc-deduplication" class="nav-link" data-scroll-target="#deduplication">Deduplication</a></li>
  </ul></li>
  <li><a href="#methylation-quantification" id="toc-methylation-quantification" class="nav-link" data-scroll-target="#methylation-quantification">Methylation Quantification</a>
  <ul class="collapse">
  <li><a href="#what-to-expect" id="toc-what-to-expect" class="nav-link" data-scroll-target="#what-to-expect">What to expect</a></li>
  <li><a href="#methylation-extraction" id="toc-methylation-extraction" class="nav-link" data-scroll-target="#methylation-extraction">Methylation extraction</a></li>
  </ul></li>
  <li><a href="#summary-report" id="toc-summary-report" class="nav-link" data-scroll-target="#summary-report">Summary Report</a></li>
  <li><a href="#file-conversions" id="toc-file-conversions" class="nav-link" data-scroll-target="#file-conversions">File Conversions</a></li>
  <li><a href="#next-steps" id="toc-next-steps" class="nav-link" data-scroll-target="#next-steps">Next Steps</a>
  <ul class="collapse">
  <li><a href="#removing-c-to-t-c-t-snps" id="toc-removing-c-to-t-c-t-snps" class="nav-link" data-scroll-target="#removing-c-to-t-c-t-snps">Removing C to T (C-&gt;T) SNPs</a></li>
  <li><a href="#downstream-application" id="toc-downstream-application" class="nav-link" data-scroll-target="#downstream-application">Downstream application</a></li>
  </ul></li>
  <li><a href="#refs" id="toc-refs" class="nav-link" data-scroll-target="#refs">References</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/sr320/course-ASI-2024/edit/main/modules/DRAFT_DNA_methylation.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/sr320/course-ASI-2024/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">DNA Methylation Assessment</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="introduction" class="level1">
<h1>Introduction</h1>
<section id="background" class="level2">
<h2 class="anchored" data-anchor-id="background">Background</h2>
<p>Thanks to advances in sample preparation and sequencing methods, there is an influx of research examining epigenomics in non-model systems. Broadly, epigenetics can be defined as changes to gene expression that do not arise from changes in the DNA sequence. DNA methylation, or addition of a methyl (CH<sub>3</sub>) group to a cytosine base adjacent to a guanine (CpG) is one of the more commonly studies epigenetic mechanisms, partly because of its environmental sensitivity and potential role in phenotypic plasticity. For a good review of epigenetic mechanisms including DNA methylation in marine organisms see <a href="#refs"><span class="citation" data-cites="doi:10.1146/annurev-marine-010318-095114">Eirin-Lopez and Putnam (<span>2019</span>)</span></a>.</p>
<p>The purpose of this tutorial is to provide an example of aligning bisulfite-treated and enzymatically converted DNA sequence data to an existing genome, and discuss needs for potential downstream applications. The principle behind creating bisulfite-treated DNA libraries is that when cytosines are not methylated, then they will be converted to uracil, which ultimately result in thymine nucleotides in the sequence data. <a href="#refs"><span class="citation" data-cites="https://doi.org/10.1111/1755-0998.13542">Wanamaker et al. (<span>2022</span>)</span></a> offers a comparison of three methods for quantifying DNA methylation at single base-pair resolution using whole genome bisulfite sequencing (WGBS), reduced representation bisulfite sequencing (RRBS), and methyl-CpG binding domain bisulfite sequencing (MBDBS). When aligning bisulfite-converted data to a reference genome the percent methylation at a given CpG loci is determined by examining the ratio of thymines to cytosines. For example if there is a given cytosine adjacent to a guanine (ie CpG locus) with 30% of the reads containing thymine, one would consider this cytosine loci (CpG) to be 70% methylated. Since bisulfite treatment can damage DNA, newer library preparation and sequencing methods (ex. <a href="https://www.neb.com/en-us/products/e7120-nebnext-enzymatic-methyl-seq-kit#Protocols,%20Manuals%20&amp;%20Usage">NEBNext® Enzymatic Methyl-seq Kit; EM-seq</a> are starting to gain popularity. As EM-Seq produces loci-level data, alignment of EM-Seq data is similar to WGBS. Also, technologies like Nanopore and PacBio sequencing can directly detect DNA methylation without bisulfite conversion. See <a href="#refs"><span class="citation" data-cites="10.1093/g3journal/jkab148">Dimond, Nguyen, and Roberts (<span>2021</span>)</span></a> as example of using Nanopore sequencing. As DNA passes through a nanopore or is sequenced by single-molecule real-time (SMRT) technology, changes in electrical current or fluorescence patterns can indicate the presence of methylated bases.</p>
</section>
<section id="experimental-considerations" class="level2">
<h2 class="anchored" data-anchor-id="experimental-considerations">Experimental Considerations</h2>
<section id="sample-size" class="level3">
<h3 class="anchored" data-anchor-id="sample-size">Sample Size</h3>
<p>Sample size is a critical factor when conducting DNA methylation analysis, particularly because of the significant variation in methylation patterns both between individuals and across different tissues. This variation, which is not yet fully understood, can significantly impact the outcomes and interpretations of methylation studies.</p>
<p>When selecting samples for DNA methylation analysis, the key principle is that larger sample sizes generally provide more reliable and generalizable results. A larger number of samples allows for a more accurate representation of the population, accounting for individual and tissue-level differences. This is essential for identifying true biological patterns rather than artifacts of small, non-representative samples.</p>
<p>However, the practicality of obtaining a large number of samples must be balanced with resource constraints. In this context, a sample size of around eight can be considered a decent starting point. This number is often sufficient to begin observing trends and patterns in methylation across individuals and tissues, providing a foundational understanding which can be built upon with further research.</p>
<p>It’s crucial to remember that while more samples can enhance the reliability of the results, the quality of the samples and the robustness of the methylation analysis techniques are equally important. Therefore, researchers must strive for a balance between quantity and quality in their sample selection to ensure meaningful and accurate insights into DNA methylation.</p>
</section>
<section id="sequencing-depth" class="level3">
<h3 class="anchored" data-anchor-id="sequencing-depth">Sequencing Depth</h3>
</section>
<section id="bisulfite-conversion-efficiency" class="level3">
<h3 class="anchored" data-anchor-id="bisulfite-conversion-efficiency">Bisulfite Conversion Efficiency</h3>
<p>Bisulfite Conversion Efficiency is important to assess. One common step during library preparation us spiking library with Lambda phage DNA. Lambda phage DNA lacks cytosine methylation, making it an ideal control for bisulfite treatment. Here’s how it works and why it’s useful:</p>
<ol type="1">
<li><p><strong>Background on Lambda Phage DNA</strong>: Lambda phage DNA is a widely used control in bisulfite sequencing because it is naturally unmethylated. By including this DNA in your sample, you create a known unmethylated reference.</p></li>
<li><p><strong>Preparation and Spiking</strong>: You prepare the lambda phage DNA and spike it into your DNA samples before bisulfite treatment. The proportion of lambda DNA added should be enough to be detected post-sequencing, but not so much that it overwhelms the sample. A typical proportion might be around 0.1% to 1% of the total DNA.</p></li>
<li><p><strong>Bisulfite Treatment and Sequencing</strong>: After spiking, you proceed with bisulfite treatment and sequencing of the entire sample, which now includes your genomic DNA of interest and the spiked lambda DNA.</p></li>
<li><p><strong>Assessing Conversion Efficiency</strong>: Once sequencing is complete, you analyze the lambda phage DNA sequences. Since this DNA is unmethylated, all cytosines should be converted to thymines after bisulfite treatment and subsequent PCR. By calculating the percentage of cytosines that were successfully converted to thymines in the lambda DNA, you can determine the bisulfite conversion efficiency. An efficient conversion typically results in over 99% of cytosines being converted.</p></li>
</ol>
</section>
</section>
<section id="tutorial-details" class="level2">
<h2 class="anchored" data-anchor-id="tutorial-details">Tutorial Details</h2>
<p>For a good review of epigenetic mechanisms including DNA methylation in marine organisms see <a href="#refs"><span class="citation" data-cites="doi:10.1146/annurev-marine-010318-095114">Eirin-Lopez and Putnam (<span>2019</span>)</span></a>. <a href="#refs"><span class="citation" data-cites="https://doi.org/10.1111/1755-0998.13542">Wanamaker et al. (<span>2022</span>)</span></a> offers a comparison of three methods for quantifying DNA methylation at single base-pair resolution using whole genome bisulfite sequencing (WGBS), reduced representation bisulfite sequencing (RRBS), and methyl-CpG binding domain bisulfite sequencing (MBDBS). The tutorial below is based on WGBS, though the general workflow would be consistent.</p>
<p>The tutorial will cover the following:<br>
- Sequence quality assessment<br>
- Read alignment<br>
- Methylation quantification<br>
- File conversions<br>
- Potential downstream applications. Although this tutorial will not provide code for downstream applications such as differential methylation analysis, it will link to other open-access resources and examples.</p>
<p>The tutorial below is based on WGBS from a species of marine intertebrate (<em>Montipora capitata</em> coral) examined in <a href="#refs"><span class="citation" data-cites="https://doi.org/10.1111/1755-0998.13542">Wanamaker et al. (<span>2022</span>)</span></a>, and all raw data can be accessed under NCBI Bioproject PRJNA691891 if you would like to follow the tutorial with this dataset. The general workflow would be consistent for other library preparation and sequencing methods that produce loci-level data. The tutorial indicates places where special analytical consideration should be given to other data types. Given the size of the data files and the computational resources necessary this is not developed to be actionable and reproducible. For context this process outlined below took weeks on a university HPC system.</p>
</section>
</section>
<section id="sequence-quality" class="level1">
<h1>Sequence Quality</h1>
<p>Following trimming one would expect to see FastQC per base sequencing content similar to the plot shown below. This particular data set was trimmed with the following parameters:</p>
<pre><code>/gscratch/srlab/programs/TrimGalore-0.4.5/trim_galore \
--output_dir /gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD \
--paired \
--fastqc_args \
--outdir /gscratch/scrubbed/strigg/analyses/20200311/WGBS_MBD/FASTQC \
--threads 28 \
--illumina \
--clip_R1 10 \
--clip_R2 10 \
--three_prime_clip_R1 10 \
--three_prime_clip_R2 10 \
--path_to_cutadapt /gscratch/srlab/programs/miniconda3/bin/cutadapt \
/gscratch/scrubbed/sr320/froger-raw/00_fastq/Meth17_R1_001.fastq.gz \</code></pre>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DRAFT_DNA_methylation/meth-bp-chart.png" class="img-fluid figure-img" style="width:70.0%"></p>
</figure>
</div>
</div>
</div>
<p>For WGBS and MBDBS data, it is general practice to hard-trim 10 bp from the beginning and end of each read. RRBS data often needs different trimming specifications <a href="https://felixkrueger.github.io/Bismark/">see Manual for details</a></p>
</section>
<section id="read-alignment" class="level1">
<h1>Read Alignment</h1>
<section id="software-considerations" class="level2">
<h2 class="anchored" data-anchor-id="software-considerations">Software Considerations</h2>
<p>The software used as part of this tutorial is <strong>Bismark Bisulfite Mapper</strong> <a href="#refs"><span class="citation" data-cites="10.1093/bioinformatics/btr167">Krueger and Andrews (<span>2011</span>)</span></a>, which is one of the more widely-used softwares to align methylation data to a reference genome. Another commonly-used options include bwa-meth <a href="#refs"><span class="citation" data-cites="pedersen2014fast">Pedersen et al. (<span>2014</span>)</span></a> and BSMAP <a href="#refs"><span class="citation" data-cites="10.1186/1471-2105-10-232">Xi and Li (<span>2009</span>)</span></a> (see <a href="#refs"><span class="citation" data-cites="10.1093/bib/bbab021">Nunn et al. (<span>2021</span>)</span></a> and <span class="citation" data-cites="10.1016/j.csbj.2022.08.051">(<a href="#ref-10.1016/j.csbj.2022.08.051" role="doc-biblioref">Gong et al. 2022</a>)</span> for comparisons of these software for plant and mammal species, respectively). As with any software it is best to be famililar with the <a href="https://felixkrueger.github.io/Bismark/">User Manual</a>.</p>
<p>Within Bismark, the tutorial uses Bowtie 2 <a href="#refs"><span class="citation" data-cites="10.1038/nmeth.1923">Langmead and Salzberg (<span>2012</span>)</span></a> as the alignment software. Although the Bismark user manual covers the necessary Bowtie 2 parameters, it may be useful to refer to the <a href="https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml">Bowtie 2 manual</a>.</p>
<p>The content below will provided with the assumption that the reader has read the manual and is meant to serve as guidance based on experience working with marine invertebrates.</p>
</section>
<section id="genome-preparation" class="level2">
<h2 class="anchored" data-anchor-id="genome-preparation">Genome Preparation</h2>
<p>The first step in the process in preparing the reference genome by performing an in silico bisulfite conversion to allow Bowtie alignments. Example code:</p>
<pre><code># Directories and programs
bismark_dir="/programs/Bismark-0.21.0"
bowtie2_dir="/programs/bowtie2-2.3.4.1-linux-x86_64/"
genome_folder="/where/the/fastafile/lives/"

${bismark_dir}/bismark_genome_preparation \
--verbose \
--parallel 28 \
--path_to_aligner ${bowtie2_dir} \
${genome_folder}</code></pre>
<ol type="1">
<li><p><strong>Bismark Genome Preparation Command</strong>:</p>
<ul>
<li><p><strong><code>${bismark_dir}/bismark_genome_preparation</code></strong>: This is the command to run the genome preparation part of Bismark. The <strong><code>${bismark_dir}</code></strong> variable is expanded to the path where Bismark is installed, so the script knows where to find the <strong><code>bismark_genome_preparation</code></strong> program.</p></li>
<li><p><strong><code>--verbose</code></strong>: This flag makes the program output more detailed information about what it is doing, which is helpful for debugging or understanding the process.</p></li>
<li><p><strong><code>--parallel 28</code></strong>: This option tells Bismark to use 28 threads in parallel to speed up the process. You can adjust this based on your computing power.</p></li>
<li><p><strong><code>--path_to_aligner ${bowtie2_dir}</code></strong>: This specifies the path to the aligner (Bowtie2) that Bismark will use. The <strong><code>${bowtie2_dir}</code></strong> variable is expanded to the path where Bowtie2 is installed.</p></li>
<li><p><strong><code>${genome_folder}</code></strong>: Finally, this specifies the location of the genome files. The script uses the <strong><code>${genome_folder}</code></strong> variable, which holds the path to these files.</p></li>
</ul></li>
</ol>
<p>Bismark will create two individual folders within this directory, one for a C-&gt;T converted genome and the other one for the G-&gt;A converted genome. After creating C-&gt;T and G-&gt;A versions of the genome they will be indexed in parallel usingthe indexer. Bismark will create the following subdirectories as outputs.</p>
<pre><code>./Bisulfite_Genome
./Bisulfite_Genome/GA_conversion
./Bisulfite_Genome/CT_conversion</code></pre>
<p>This preparation step only needs to happen once for a given genome.</p>
</section>
<section id="aligning-reads" class="level2">
<h2 class="anchored" data-anchor-id="aligning-reads">Aligning reads</h2>
<p>Once the genome is prepared, reads can be aligned. For aligning the trimmed reads to the genome the following code structure is used:</p>
<pre><code>find ${reads_dir}*_R1_001_val_1.fq.gz \
| xargs basename -s _R1_001_val_1.fq.gz | xargs -I{} ${bismark_dir}/bismark \
--path_to_bowtie ${bowtie2_dir} \
-genome ${genome_folder} \
-p 4 \
-score_min L,0,-0.6 \
--non_directional \
-1 ${reads_dir}{}_R1_001_val_1.fq.gz \
-2 ${reads_dir}{}_R2_001_val_2.fq.gz \
-o Mcap_tg</code></pre>
<p>This will create BAM files (sequence alignment files) and txt files (mapping reports). The reports contain information such as mapping efficiency.</p>
<p>Specifically this is what the code chunk does:</p>
<ol type="1">
<li><p><strong>Finding Files and Preparing Filenames</strong>:</p>
<ul>
<li><p><strong><code>find ${reads_dir}*_R1_001_val_1.fq.gz</code></strong>: This command uses the <strong><code>find</code></strong> utility to search for files in the directory specified by the variable <strong><code>reads_dir</code></strong>. The files it looks for have a naming pattern ending with <strong><code>_R1_001_val_1.fq.gz</code></strong>, which typically denotes the first file of a pair in paired-end sequencing data.</p></li>
<li><p><strong><code>| xargs basename -s _R1_001_val_1.fq.gz</code></strong>: The <strong><code>xargs</code></strong> command takes the list of files found and passes them to <strong><code>basename</code></strong>, which strips the suffix <strong><code>_R1_001_val_1.fq.gz</code></strong> from each filename. This step extracts the base name of each file, which is used to pair the R1 and R2 files correctly in the following steps.</p></li>
</ul></li>
<li><p><strong>Running Bismark for Sequence Alignment</strong>:</p>
<ul>
<li><p><strong><code>| xargs -I{} ${bismark_dir}/bismark</code></strong>: The <strong><code>xargs</code></strong> command here is used to pass the base filenames to Bismark for alignment. <strong><code>{}</code></strong> is a placeholder for each base filename.</p></li>
<li><p><strong><code>--path_to_bowtie ${bowtie2_dir}</code></strong>: This option specifies the path to the Bowtie 2 aligner, which Bismark uses for the alignment process.</p></li>
<li><p><strong><code>-genome ${genome_folder}</code></strong>: Specifies the directory where the genome files, prepared in advance, are located.</p></li>
<li><p><strong><code>-p 4</code></strong>: Sets the number of processing threads to 4, allowing Bismark to perform the alignment in parallel, which can speed up the process. This parameter can be changed based on your computing power.</p></li>
<li><p><strong><code>-score_min L,0,-0.6</code></strong>: This sets the scoring threshold for alignments, impacting the sensitivity and specificity of the alignment. The default scoring threshold used by Bismark is L,0,-0.2.</p></li>
<li><p><strong><code>--non_directional</code></strong>: This flag indicates the sequencing data is non-directional, which is important for correctly interpreting the methylation status in bisulfite sequencing.</p></li>
<li><p><strong><code>-1 ${reads_dir}{}_R1_001_val_1.fq.gz -2 ${reads_dir}{}_R2_001_val_2.fq.gz</code></strong>: Specifies the files for paired-end reads. R1 and R2 files are paired based on their base filename.</p></li>
<li><p><strong><code>-o Mcap_tg</code></strong>: Sets the output directory for the alignment results to <strong><code>Mcap_tg</code></strong>.</p></li>
</ul></li>
</ol>
<section id="optimizing-alignment-sensitivity" class="level3">
<h3 class="anchored" data-anchor-id="optimizing-alignment-sensitivity">Optimizing alignment sensitivity</h3>
<p>The code above uses <code>score_min L,0,-0.6</code> as the scoring threshold for alignment sensitivity and specificity. This is different than the Bismark default, <code>score_min L,0,-0.2</code>. Reducing the sensitivity of the alignment can increase the percentage of reads aligned to the reference genome, but of course the quality of the alignment may decrease. A good practice is to align a subset of your data to the reference genome using different <code>score_min</code> parameters, then comparing mapping efficiency from the mapping reports. Generally, you want to choose a parameter that maximizes percent alignment and specificity. Example code</p>
<pre><code>find ${reads_dir}*_R1_001_val_1.fq.gz \
| xargs basename -s _R1_001_val_1.fq.gz | xargs -I{} ${bismark_dir}/bismark \
--path_to_bowtie ${bowtie2_dir} \
-genome ${genome_folder} \
-p 4 \
- u 10000 \
-score_min L,0,-0.6 \
--non_directional \
-1 ${reads_dir}{}_R1_001_val_1.fq.gz \
-2 ${reads_dir}{}_R2_001_val_2.fq.gz \
-o Mcap_tg</code></pre>
<p>This is the same code as above, except <code>-u 10000</code> is used to run the alignment on a 10,000 bp subset of the data in each input file.</p>
</section>
</section>
<section id="deduplication" class="level2">
<h2 class="anchored" data-anchor-id="deduplication">Deduplication</h2>
<p>Files are then deduplicated if they are whole genome bisulfite treated samples. This command will deduplicate the Bismark alignment BAM file and remove all reads that align to the the very same position and in the same orientation except for one. This step is recommended for whole-genome bisulfite samples, but <strong>should not</strong> be used for reduced representation libraries such as RRBS, amplicon, or target enrichment libraries. The <a href="https://rawgit.com/FelixKrueger/Bismark/master/Docs/Bismark_User_Guide.html#ix-notes-about-different-library-types-and-commercial-kits"><code>bismark</code> manual</a> provides additional information.</p>
<pre><code>find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${bismark_dir}/deduplicate_bismark \
--bam \
--paired \
{}.bam</code></pre>
</section>
</section>
<section id="methylation-quantification" class="level1">
<h1>Methylation Quantification</h1>
<section id="what-to-expect" class="level2">
<h2 class="anchored" data-anchor-id="what-to-expect">What to expect</h2>
<p>DNA methylation at CpG locus can vary to 0-100%. How methylated CpGs are across a genome can vary WIDELY depending on your focal taxa. Generally, invertebrate genomes are predominantly unmethylated, with smaller regions of higher methylation in a mosaic pattern. In some marine invertebrates we see ~15% methylation of CpGs (defined as &gt; 75% methylated) across the genome with most CpGs unmethylated. In these taxa methylation usually occurs in gene bodies.</p>
<p>The figure below shows the methylation ratio of CpGs across the Pacific oyster (<em>Crassostrea gigas</em>) genome, as well as the location of methylation across the genome:</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DRAFT_DNA_methylation/meth_ratio.png" class="img-fluid figure-img" style="width:70.0%"></p>
<figcaption class="figure-caption">Frequency distribution of methylation ratios for CpG dinucleotides in oyster gill tissue. <a href="#refs"><span class="citation" data-cites="10.7717/peerj.215">Gavery and Roberts (<span>2013</span>)</span></a></figcaption>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DRAFT_DNA_methylation/igv.png" class="img-fluid figure-img" style="width:100.0%"></p>
<figcaption class="figure-caption">Screenshot from IGV. Bedgraph files are displayed. See below for code to create these file (File Conversion)</figcaption>
</figure>
</div>
</div>
</div>
<hr>
<p>Vertebrate genomes, on the other hand, have higher overall levels of methylation, with methylation occuring in CpG islands. An example analysis of methylation at individual CG dinucleotides in an elephant shark, mouse, and zebra fish is shown, below. Results reavel comparable levels of global methyaltion patterns with the majority of sites being highly methylated (≥ 80%), and fewer than one tenth being unmethylated (doi: 10.12688/f1000research.11281.1).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DRAFT_DNA_methylation/vert_methyl_distr.png" class="img-fluid figure-img" style="width:70.0%"></p>
<figcaption class="figure-caption">Genome screenshots of methylation quantified in 2kb running windows over the first 3Mb of chromosome 1 zebrafish and mouse, and of the largest scaffold (NW_006890054.1) in elephant shark (doi: 10.12688/f1000research.11281.1)</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="methylation-extraction" class="level2">
<h2 class="anchored" data-anchor-id="methylation-extraction">Methylation extraction</h2>
<p>Methylation levels are then extracted using the <code>bismark_methylation_extractor</code>. For example:</p>
<pre><code>${bismark_dir}/bismark_methylation_extractor \
--bedGraph \
--counts \
--comprehensive \
--merge_non_CpG \
--multicore 28 \
--buffer_size 75% \
*deduplicated.bam</code></pre>
<p>This particular command is used to extract and process methylation data from BAM files (generated after sequencing data alignment). Let’s break down the options:</p>
<ol type="1">
<li><p><strong>Options</strong>:</p>
<ul>
<li><p><strong><code>--bedGraph</code></strong>: Generates a BedGraph file output, which is a format that allows visualization of the methylation levels across the genome in genome browsers.</p></li>
<li><p><strong><code>--counts</code></strong>: Outputs the count of methylated and unmethylated calls for each cytosine in the genome.</p></li>
<li><p><strong><code>--comprehensive</code></strong>: This option makes the methylation extractor process all cytosines in the genome, not just those in a CpG context. It combines all three contexts (CpG, CHG, and CHH) into one output file.</p></li>
<li><p><strong><code>--merge_non_CpG</code></strong>: Merges the methylation information for CHG and CHH contexts (non-CpG) into a single file.</p></li>
<li><p><strong><code>--multicore 28</code></strong>: Specifies that the program should use 28 cores to process the data in parallel, which can significantly speed up the analysis. This can be adjusted depending on your computing power.</p></li>
<li><p><strong><code>--buffer_size 75%</code></strong>: This option sets the buffer size for reading in BAM files to 75% of the available system memory. It helps optimize the performance of the tool by adjusting how much memory is used for processing. This can be adjusted depending on your computing power.</p></li>
<li><p><strong><code>*deduplicated.bam</code></strong>: This is the input file pattern for the command. It indicates that the methylation extractor should process all BAM files in the current directory that have names ending with “deduplicated.bam”. These files are typically the output of a deduplication process applied to aligned bisulfite sequencing data.</p></li>
</ul></li>
</ol>
<p>This will create a file with the suffix <code>deduplicated.bismark.cov.gz</code> (example shown below in uncompressed format):</p>
<pre><code>NC_035784.1 141 141 37.5    3   5
NC_035784.1 142 142 100 2   0
NC_035784.1 155 155 70  7   3
NC_035784.1 156 156 100 2   0
NC_035784.1 291 291 0   0   2
NC_035784.1 292 292 0   0   3
NC_035784.1 313 313 0   0   1
NC_035784.1 314 314 66.6666666666667    2   1
NC_035784.1 470 470 66.6666666666667    4   2
NC_035784.1 611 611 0   0   4</code></pre>
<p>column organization of the file</p>
<pre><code>&lt;chromosome&gt; &lt;start position&gt; &lt;end position&gt; &lt;methylation percentage&gt; &lt;count methylated&gt; &lt;count unmethylated&gt;</code></pre>
<p>The resultant bedGraph file contains information from <em>unmerged</em> strands, meaning reads aligning to forward and reverse strands in the same position are not combined. It is important to merge the reads from the forward and reserve strands for more accurate methylation quantification. A genome-wide cytosine report and merged bedGraph file are generated (including merged bedGraph file) from the <code>deduplicated.bismark.cov.gz</code> generated above in this example code:</p>
<pre><code>find *deduplicated.bismark.cov.gz \
| xargs basename -s _trimmed_bismark_bt2.deduplicated.bismark.cov.gz \
| xargs -I{} ${bismark_dir}/coverage2cytosine \
--genome_folder ${genome_folder} \
-o {} \
--merge_CpG \
--zero_based \
{}_trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></pre>
<p>The script converts coverage files into a format that details cytosine methylation levels across the genome. Here’s a breakdown of each step:</p>
<ol type="1">
<li><p><strong>Finding Files</strong>:</p>
<ul>
<li><strong><code>find *deduplicated.bismark.cov.gz</code></strong>: This command is used to search for all files in the current directory that end with <strong><code>deduplicated.bismark.cov.gz</code></strong>. These files are coverage files produced by Bismark, typically after aligning bisulfite-treated DNA sequences. They contain information about methylation at different sites in the genome.</li>
</ul></li>
<li><p><strong>Extracting Base File Names</strong>:</p>
<ul>
<li><strong><code>| xargs basename -s _trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></strong>: This pipeline uses <strong><code>xargs</code></strong> to pass each found file to <strong><code>basename</code></strong>, which strips the suffix <strong><code>_trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></strong> from each file name. This step extracts the base names of the files, which will be used to generate output filenames in the following steps.</li>
</ul></li>
<li><p><strong>Converting Coverage to Cytosine Methylation Levels</strong>:</p>
<ul>
<li><p><strong><code>| xargs -I{} ${bismark_dir}/coverage2cytosine</code></strong>: This pipeline uses <strong><code>xargs</code></strong> again to pass each base filename to the <strong><code>coverage2cytosine</code></strong> tool in the Bismark suite. The placeholder <strong><code>{}</code></strong> is replaced by the base filename.</p></li>
<li><p><strong><code>--genome_folder ${genome_folder}</code></strong>: This option specifies the location of the genome folder. The <strong><code>${genome_folder}</code></strong> variable holds the path to this folder, which contains reference genome data necessary for the analysis.</p></li>
<li><p><strong><code>-o {}</code></strong>: The <strong><code>-o</code></strong> option specifies the output file name, using the base filename <strong><code>{}</code></strong> obtained from the previous steps.</p></li>
<li><p><strong><code>--merge_CpG</code></strong>: This flag instructs the program to merge the methylation information of cytosines in CpG context, providing a single methylation value for each CpG site.</p></li>
<li><p><strong><code>--zero_based</code></strong>: This option indicates that the output file should use zero-based coordinates, which is a common format in bioinformatics where the first base of a sequence is numbered 0 instead of 1.</p></li>
<li><p><strong><code>{}_trimmed_bismark_bt2.deduplicated.bismark.cov.gz</code></strong>: Finally, this specifies the input file for each iteration, where <strong><code>{}</code></strong> is replaced by the base filename obtained earlier. These are the coverage files previously identified by the <strong><code>find</code></strong> command.</p></li>
</ul></li>
</ol>
<p>Output file format <code>.CpG_report.merged_CpG_evidence.cov</code> (merged bedGraph file)</p>
<pre><code>NC_035785.1 217 219 100.000000  17  0
NC_035785.1 523 525 87.500000   7   1
NC_035785.1 556 558 50.000000   5   5
NC_035785.1 727 729 100.000000  16  0
NC_035785.1 1330    1332    0.000000    0   2
NC_035785.1 1403    1405    0.000000    0   2
NC_035785.1 1494    1496    66.666667   2   1
NC_035785.1 1747    1749    100.000000  8   0
NC_035785.1 2024    2026    100.000000  24  0
NC_035785.1 2054    2056    93.333333   14  1</code></pre>
<p>column organization of the file</p>
<pre><code>&lt;chromosome&gt; &lt;start position&gt; &lt;end position&gt; &lt;methylation percentage&gt; &lt;count methylated&gt; &lt;count unmethylated&gt;</code></pre>
</section>
</section>
<section id="summary-report" class="level1">
<h1>Summary Report</h1>
<p>Many steps of the Bismark workflow produce reports (.txt files) with useful information (ex. mapping efficiency, percent cytosine methylation, etc.). These files can be collated into processing reports for each individual sample, which can then be further collated into an <a href="https://gannet.fish.washington.edu/seashell/bu-mox/scrubbed/032120-fds/bismark_summary_report.html">HTML summary report</a>:</p>
<pre><code>#HTML Processing Report

${bismark_dir}/bismark2report

#Summary Report

${bismark_dir}/bismark2summary</code></pre>
<p>These commands should be run in the same directory where all output files are stored.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="DRAFT_DNA_methylation/bismark-summary-report.png" class="img-fluid figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="file-conversions" class="level1">
<h1>File Conversions</h1>
<p>From here it could be useful to do some file reorganizaion to obtain bedGraph or tab files for downstream analysis (eg <code>bedtools</code>, GLMs). Specifically, it may be important to filter the CpG methylation data to a specific coverage threshold (ex. 5x or 10x). The more read coverage a CpG has, the higher the confidence in methylation quantification.</p>
<p>Creating bedGraphs post-merge at a specific coverage threshold (ex. 10x or 5x):</p>
<pre><code>for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 &gt;= 5) {print $1, $2, $3, $4}}' \
  &gt; "${STEM}"_5x.bedgraph
done</code></pre>
<p>Creating tab files with raw count data for GLMs at 10x and 5x coverage:</p>
<pre><code>for f in *merged_CpG_evidence.cov
do
  STEM=$(basename "${f}" .CpG_report.merged_CpG_evidence.cov)
  cat "${f}" | awk -F $'\t' 'BEGIN {OFS = FS} {if ($5+$6 &gt;= 5) {print $1, $2, $3, $4, $5, $6}}' \
  &gt; "${STEM}"_5x.tab
done</code></pre>
<p>It is sometimes useful to sort BAM files for downstream analysis (eg methylkit, IGV)</p>
<pre><code># Sort files for methylkit and IGV
find *.bam | \
xargs basename -s .bam | \
xargs -I{} ${samtools} \
sort --threads 28 {}.bam \
-o {}.sorted.bam

# Index sorted files for IGV
find *.sorted.bam | \
xargs basename -s .sorted.bam | \
xargs -I{} ${samtools} \
index -@ 28 {}.sorted.bam</code></pre>
</section>
<section id="next-steps" class="level1">
<h1>Next Steps</h1>
<p>So you’ve aligned your trimmed reads to your reference genome and have methylation data for CpGs. Now what?</p>
<section id="removing-c-to-t-c-t-snps" class="level2">
<h2 class="anchored" data-anchor-id="removing-c-to-t-c-t-snps">Removing C to T (C-&gt;T) SNPs</h2>
<p>Removing cytosine (C) to thymine (T) single nucleotide polymorphisms (SNPs) in bisulfite sequencing analysis could be something to consider for a few reasons:</p>
<ol type="1">
<li><p><strong>Bisulfite Conversion Process</strong>: In bisulfite sequencing, unmethylated cytosines are converted to uracil, which is then read as thymine in sequencing. This conversion is key to identifying methylated cytosines. However, if a genomic DNA sequence naturally has a C to T (C-&gt;T) SNP, it can be falsely interpreted as a converted cytosine, leading to incorrect conclusions about methylation.</p></li>
<li><p><strong>Reduction of False Positives</strong>: By removing C-&gt;T SNPs, one can reduce false positives in methylation analysis. Without removal, these SNPs might be misinterpreted as unmethylated cytosines that have undergone conversion, skewing the data towards an apparent increase in methylation.</p></li>
<li><p><strong>Accurate Methylation Mapping</strong>: For precise mapping of methylation patterns, it’s crucial to distinguish between true methylation signals and genetic variation (SNPs). Removing C-&gt;T SNPs allows for a more accurate representation of the methylation status of the genome.</p></li>
<li><p><strong>Analysis Consistency</strong>: In comparative studies, where multiple samples or conditions are analyzed, removing C-&gt;T SNPs ensures consistency across samples. This is especially important in cases where SNP profiles may vary between samples, which could otherwise lead to inconsistent methylation data.</p></li>
</ol>
<p>By eliminating these potential sources of error, researchers can have more confidence in their interpretation of the bisulfite sequencing data, leading to more reliable and valid conclusions about DNA methylation patterns.</p>
<p>One way to do this is using BS-Snper (<span class="citation" data-cites="10.1093/bioinformatics/btv507">Gao et al. (<a href="#ref-10.1093/bioinformatics/btv507" role="doc-biblioref">2015</a>)</span>). <a href="https://github.com/hellbelly/BS-Snper" class="uri">https://github.com/hellbelly/BS-Snper</a>. This program finds SNPs in Bisulfite treated alignment files. The bash code below represents steps to identify SNPS and create a list of all C-&gt;T SNPS.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode perl code-with-copy"><code class="sourceCode perl"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>samtools merge \</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>Pg_merged.bam \</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="wa">*.</span>sorted.bam</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>perl /gscratch/srlab/programs/BS<span class="ot">-S</span>nper-master/BS<span class="ot">-S</span>nper.pl \</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>Pg_merged.bam \</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>-<span class="ot">-f</span>a Panopea<span class="ot">-g</span>enerosa-v1.<span class="dv">0</span>.fa \</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>-<span class="ot">-o</span>utput snp.candidate.out \</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>--methcg meth.cg \</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>--methchg meth.chg \</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>--methchh meth.chh \</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>--minhetfreq <span class="fl">0.1</span> \</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>--minhomfreq <span class="fl">0.85</span> \</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>--minquali <span class="dv">15</span> \</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>--mincover <span class="dv">10</span> \</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>--maxcover <span class="dv">1000</span> \</span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>--minread2 <span class="dv">2</span> \</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>-<span class="ot">-e</span>rrorate <span class="fl">0.02</span> \</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>--mapvalue <span class="dv">20</span> \</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>&gt;SNP.vcf <span class="dv">2</span>&gt;SNP.<span class="fu">log</span></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="wa">$'C</span>\tG<span class="ot">'</span><span class="ss"> output/SNP.vcf &gt; output/CT-SNP.vcf</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a><span class="ss">wc -l output/CT-SNP.vcf</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s break down the command and its arguments:</p>
<ol type="1">
<li><p><strong><code>perl /gscratch/srlab/programs/BS-Snper-master/BS-Snper.pl</code></strong>: This part of the command calls the Perl interpreter to execute the <strong><code>BS-Snper.pl</code></strong> script. The script is located in the <strong><code>/gscratch/srlab/programs/BS-Snper-master/</code></strong> directory.</p></li>
<li><p><strong><code>Pg_merged.bam</code></strong>: This is the input file for the script, likely a BAM file (Binary Alignment/Map format) which contains alignment information of bisulfite-treated sequencing reads.</p></li>
<li><p><strong><code>--fa Panopea-generosa-v1.0.fa</code></strong>: This option specifies the reference genome file in FASTA format against which the BAM file will be compared.</p></li>
<li><p><strong><code>--output snp.candidate.out</code></strong>: This argument sets the name of the output file where the SNP candidates will be written.</p></li>
<li><p><strong><code>--methcg meth.cg</code>, <code>--methchg meth.chg</code>, <code>--methchh meth.chh</code></strong>: These options specify the output files for methylation analysis for CG, CHG, and CHH contexts, respectively (where H can be A, C, or T).</p></li>
<li><p><strong><code>--minhetfreq 0.1</code></strong>: Sets the minimum frequency for heterozygous bases to be considered in SNP calling (here, 10%).</p></li>
<li><p><strong><code>--minhomfreq 0.85</code></strong>: Sets the minimum frequency for homozygous bases to be considered in SNP calling (here, 85%).</p></li>
<li><p><strong><code>--minquali 15</code></strong>: Sets the minimum base quality score.</p></li>
<li><p><strong><code>--mincover 10</code>, <code>--maxcover 1000</code></strong>: These set the minimum and maximum coverage (number of reads) for a position to be considered in the analysis.</p></li>
<li><p><strong><code>--minread2 2</code></strong>: The minimum number of reads required to support a SNP candidate. It could be useful to use the same values for the minimum read threshold for SNP identification and for minimum coverage in the methylation data.</p></li>
<li><p><strong><code>--errorate 0.02</code></strong>: Sets the error rate of sequencing (here, 2%).</p></li>
<li><p><strong><code>--mapvalue 20</code></strong>: Specifies the mapping quality value, a threshold for considering reads in the analysis.</p></li>
<li><p><strong><code>&gt;SNP.vcf</code></strong>: Redirects the standard output to a file named <strong><code>SNP.vcf</code></strong>. This file will contain the SNP calls in VCF (Variant Call Format) format.</p></li>
<li><p><strong><code>2&gt;SNP.log</code></strong>: Redirects the standard error output to a file named <strong><code>SNP.log</code></strong>. This file will contain error messages and possibly additional log information.</p></li>
</ol>
<p>The resulting could be used to mask the genome and the entire process described above could be repeated (ie Genome Preparation, Alignment, Quantification). An alternative and common practice is to remove cytosines from downstream analysis by removing loci in intermediate files (ie tab-delimited data files). Below is an example using <code>antijoin</code> to remove loci that correspond to CT SNPs from 5x coverage methyation quantification files.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in CT SNP file</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>ct <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/CT-SNP.vcf"</span>, <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">loci =</span> <span class="fu">paste0</span>(V1, <span class="st">"_"</span>, V2))</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. List all files with _5x.tab suffix</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="st">"../data/"</span>, <span class="at">pattern =</span> <span class="st">"_5x.tab$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Iterate over each file</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(file <span class="cf">in</span> files) {</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract base filename without the directory for naming purposes</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  base_name <span class="ot">&lt;-</span> <span class="fu">basename</span>(file)</span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Read the file</span></span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(file, <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Modify the data</span></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a>  modified_data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">loci =</span> <span class="fu">paste0</span>(V1, <span class="st">"_"</span>, V2)) <span class="sc">%&gt;%</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">anti_join</span>(ct, <span class="at">by =</span> <span class="st">"loci"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>loci)</span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Write the modified data to an output file</span></span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>  output_file <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"../output/f"</span>, base_name)</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write.table</span>(modified_data, <span class="at">file =</span> output_file, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s break down what each part of the script does:</p>
<ol type="1">
<li><p><strong>Reading in a CT SNP file</strong>:</p>
<ul>
<li><p><strong><code>ct &lt;- read.csv("../output/CT-SNP.vcf", header = FALSE, sep = "\t")</code></strong>: This line reads a file named <strong><code>CT-SNP.vcf</code></strong> from the specified directory. The file is expected to be in a tab-separated format (<strong><code>sep = "\t"</code></strong>), and it doesn’t have a header row (<strong><code>header = FALSE</code></strong>).</p></li>
<li><p><strong><code>mutate(loci = paste0(V1, "_", V2))</code></strong>: The <strong><code>mutate</code></strong> function from the <strong><code>dplyr</code></strong> package is used to add a new column (<strong><code>loci</code></strong>) to the <strong><code>ct</code></strong> data frame. This column is created by concatenating the values in the first (<strong><code>V1</code></strong>) and second columns (<strong><code>V2</code></strong>) of each row, separated by an underscore.</p></li>
</ul></li>
<li><p><strong>Listing files with a specific suffix</strong>:</p>
<ul>
<li><strong><code>files &lt;- list.files(path = "../data/", pattern = "_5x.tab$", full.names = TRUE)</code></strong>: This line generates a list of file paths in the <strong><code>../data/</code></strong> directory. It only includes files that end with <strong><code>_5x.tab</code></strong> (as indicated by the <strong><code>pattern</code></strong> argument).</li>
</ul></li>
<li><p><strong>Processing each file</strong>:</p>
<ul>
<li><p>The <strong><code>for(file in files)</code></strong> loop iterates over each file found in the previous step.</p>
<ul>
<li><p><strong><code>base_name &lt;- basename(file)</code></strong>: Extracts the file name without the directory path.</p></li>
<li><p><strong><code>data &lt;- read.csv(file, header = FALSE, sep = "\t")</code></strong>: Reads the current file in the loop. Similar to the first <strong><code>read.csv</code></strong> call, it expects a tab-separated file with no header row.</p></li>
<li><p>The <strong><code>mutate</code></strong> and <strong><code>anti_join</code></strong> functions are used to modify <strong><code>data</code></strong>:</p>
<ul>
<li><p><strong><code>mutate(loci = paste0(V1, "_", V2))</code></strong>: Adds a <strong><code>loci</code></strong> column similar to the <strong><code>ct</code></strong> data frame.</p></li>
<li><p><strong><code>anti_join(ct, by = "loci")</code></strong>: Removes rows from <strong><code>data</code></strong> that have matching <strong><code>loci</code></strong> values in the <strong><code>ct</code></strong> data frame.</p></li>
<li><p><strong><code>select(-loci)</code></strong>: Removes the <strong><code>loci</code></strong> column from the data.</p></li>
</ul></li>
<li><p>The modified data is then written to an output file:</p>
<ul>
<li><p><strong><code>output_file &lt;- paste0("../output/f", base_name)</code></strong>: Constructs the output file path.</p></li>
<li><p><strong><code>write.table(modified_data, file = output_file, sep = "\t", row.names = FALSE, quote = FALSE, col.names = FALSE)</code></strong>: Writes the <strong><code>modified_data</code></strong> to the specified file. The output is tab-separated, and it doesn’t include row names, quotes, or column names.</p></li>
</ul></li>
</ul></li>
</ul></li>
</ol>
</section>
<section id="downstream-application" class="level2">
<h2 class="anchored" data-anchor-id="downstream-application">Downstream application</h2>
<ul>
<li><p>Integrating with population genetic data. example: <a href="#refs"><span class="citation" data-cites="10.1093/gbe/evad013">Silliman et al. (<span>2023</span>)</span></a></p></li>
<li><p>Gene-level methylation analysis: example <a href="#refs"><span class="citation" data-cites="10.1101/2022.06.24.497506">Putnam et al. (<span>2022</span>)</span></a>. It is worth considering the utility of converting loci-level resolution data to gene-level resolution data.</p></li>
<li><p>Differential loci methylation and/or region analyses. example: <a href="#refs"><span class="citation" data-cites="10.1186/s12864-022-08781-5">Venkataraman, White, and Roberts (<span>2022</span>)</span></a>. methylKit is a popular package to conduct this analysis differential methylation anlaysis</p></li>
<li><p>Feature overlap. example: <a href="#refs"><span class="citation" data-cites="10.3389/fmars.2020.00225">Venkataraman et al. (<span>2020</span>)</span></a> Bedtools is excellent for quantitative assessment of feature overlap and IGV is good for visual assessment.</p></li>
</ul>
</section>
</section>


<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><section id="refs" class="level1">
<h1>References</h1>


<div id="refs" class="references csl-bib-body hanging-indent" role="list">
<div id="ref-10.1093/g3journal/jkab148" class="csl-entry" role="listitem">
Dimond, J. L., N. Nguyen, and S. B. Roberts. 2021. <span>“Dna Methylation Profiling of a Cnidarian-Algal Symbiosis Using Nanopore Sequencing.”</span> <em>G3 Genes|Genomes|Genetics</em> 11. <a href="https://doi.org/10.1093/g3journal/jkab148">https://doi.org/10.1093/g3journal/jkab148</a>.
</div>
<div id="ref-doi:10.1146/annurev-marine-010318-095114" class="csl-entry" role="listitem">
Eirin-Lopez, Jose M., and Hollie M. Putnam. 2019. <span>“Marine Environmental Epigenetics.”</span> <em>Annual Review of Marine Science</em> 11 (1): 335–68. <a href="https://doi.org/10.1146/annurev-marine-010318-095114">https://doi.org/10.1146/annurev-marine-010318-095114</a>.
</div>
<div id="ref-10.1093/bioinformatics/btv507" class="csl-entry" role="listitem">
Gao, Shengjie, Dan Zou, Likai Mao, Huayu Liu, Pengfei Song, Youguo Chen, Shancen Zhao, et al. 2015. <span>“<span class="nocase">BS-SNPer: SNP calling in bisulfite-seq data</span>.”</span> <em>Bioinformatics</em> 31 (24): 4006–8. <a href="https://doi.org/10.1093/bioinformatics/btv507">https://doi.org/10.1093/bioinformatics/btv507</a>.
</div>
<div id="ref-10.7717/peerj.215" class="csl-entry" role="listitem">
Gavery, Mackenzie R., and Steven B. Roberts. 2013. <span>“Predominant Intragenic Methylation Is Associated with Gene Expression Characteristics in a Bivalve Mollusc.”</span> <em>PeerJ</em> 1 (November): e215. <a href="https://doi.org/10.7717/peerj.215">https://doi.org/10.7717/peerj.215</a>.
</div>
<div id="ref-10.1016/j.csbj.2022.08.051" class="csl-entry" role="listitem">
Gong, W., X. Pan, D. Xu, G. Ji, Y. Wang, Y. Tian, J. Cai, J. Li, Z. Zhang, and X. Yuan. 2022. <span>“Benchmarking Dna Methylation Analysis of 14 Alignment Algorithms for Whole Genome Bisulfite Sequencing in Mammals.”</span> <em>Computational and Structural Biotechnology Journal</em> 20: 4704–16. <a href="https://doi.org/10.1016/j.csbj.2022.08.051">https://doi.org/10.1016/j.csbj.2022.08.051</a>.
</div>
<div id="ref-10.1093/bioinformatics/btr167" class="csl-entry" role="listitem">
Krueger, Felix, and Simon R. Andrews. 2011. <span>“<span class="nocase">Bismark: a flexible aligner and methylation caller for Bisulfite-Seq applications</span>.”</span> <em>Bioinformatics</em> 27 (11): 1571–72. <a href="https://doi.org/10.1093/bioinformatics/btr167">https://doi.org/10.1093/bioinformatics/btr167</a>.
</div>
<div id="ref-10.1038/nmeth.1923" class="csl-entry" role="listitem">
Langmead, B., and S. L. Salzberg. 2012. <span>“Fast Gapped-Read Alignment with Bowtie 2.”</span> <em>Nature Methods</em> 9: 357–59. <a href="https://doi.org/10.1038/nmeth.1923">https://doi.org/10.1038/nmeth.1923</a>.
</div>
<div id="ref-10.1093/bib/bbab021" class="csl-entry" role="listitem">
Nunn, A., C. Otto, P. F. Stadler, and D. Langenberger. 2021. <span>“Comprehensive Benchmarking of Software for Mapping Whole Genome Bisulfite Data: From Read Alignment to Dna Methylation Analysis.”</span> <em>Briefings in Bioinformatics</em> 22. <a href="https://doi.org/10.1093/bib/bbab021">https://doi.org/10.1093/bib/bbab021</a>.
</div>
<div id="ref-pedersen2014fast" class="csl-entry" role="listitem">
Pedersen, Brent S., Kenneth Eyring, Subhajyoti De, Ivana V. Yang, and David A. Schwartz. 2014. <span>“Fast and Accurate Alignment of Long Bisulfite-Seq Reads.”</span> <a href="https://arxiv.org/abs/1401.1129">https://arxiv.org/abs/1401.1129</a>.
</div>
<div id="ref-10.1101/2022.06.24.497506" class="csl-entry" role="listitem">
Putnam, H. M., S. A. Trigg, S. J. White, L. H. Spencer, B. Vadopalas, A. Natarajan, J. Hetzel, et al. 2022. <span>“Dynamic Dna Methylation Contributes to Carryover Effects and Beneficial Acclimatization in Geoduck Clams.”</span> <a href="https://doi.org/10.1101/2022.06.24.497506">https://doi.org/10.1101/2022.06.24.497506</a>.
</div>
<div id="ref-10.1093/gbe/evad013" class="csl-entry" role="listitem">
Silliman, K., L. H. Spencer, S. J. White, and S. B. Roberts. 2023. <span>“Epigenetic and Genetic Population Structure Is Coupled in a Marine Invertebrate.”</span> <em>Genome Biology and Evolution</em> 15. <a href="https://doi.org/10.1093/gbe/evad013">https://doi.org/10.1093/gbe/evad013</a>.
</div>
<div id="ref-10.3389/fmars.2020.00225" class="csl-entry" role="listitem">
Venkataraman, Y. R., A. M. Downey-Wall, J. B. Ries, I. Westfield, S. J. White, S. B. Roberts, and K. E. Lotterhos. 2020. <span>“General Dna Methylation Patterns and Environmentally-Induced Differential Methylation in the Eastern Oyster (Crassostrea Virginica).”</span> <em>Frontiers in Marine Science</em> 7. <a href="https://doi.org/10.3389/fmars.2020.00225">https://doi.org/10.3389/fmars.2020.00225</a>.
</div>
<div id="ref-10.1186/s12864-022-08781-5" class="csl-entry" role="listitem">
Venkataraman, Y. R., S. J. White, and S. B. Roberts. 2022. <span>“Differential Dna Methylation in Pacific Oyster Reproductive Tissue in Response to Ocean Acidification.”</span> <em>BMC Genomics</em> 23. <a href="https://doi.org/10.1186/s12864-022-08781-5">https://doi.org/10.1186/s12864-022-08781-5</a>.
</div>
<div id="ref-https://doi.org/10.1111/1755-0998.13542" class="csl-entry" role="listitem">
Wanamaker, Shelly A., Yaamini R. Venkataraman, Mackenzie R. Gavery, Steven B. Roberts, Debashish Bhattacharya, Alan Downey-Wall, Jose M. Eirin-Lopez, et al. 2022. <span>“Invertebrate Methylomes Provide Insight into Mechanisms of Environmental Tolerance and Reveal Methodological Biases.”</span> <em>Molecular Ecology Resources</em> 22 (4): 1247–61. <a href="https://doi.org/10.1111/1755-0998.13542">https://doi.org/10.1111/1755-0998.13542</a>.
</div>
<div id="ref-10.1186/1471-2105-10-232" class="csl-entry" role="listitem">
Xi, Y., and W. Li. 2009. <span>“Bsmap: Whole Genome Bisulfite Sequence Mapping Program.”</span> <em>BMC Bioinformatics</em> 10. <a href="https://doi.org/10.1186/1471-2105-10-232">https://doi.org/10.1186/1471-2105-10-232</a>.
</div>
</div>
</section></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>